### ISSUE TITLE
Remaining clean up for migration tests

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#19668]
After #19039 is merged, we have some of the following cleanup to do:

* [ ] Go over ModelDifferTests and remove everything already covered by the new functional migration tests
* [ ] Go over remaining tests in MigrationSqlGeneratorTestBase and see if they make sense. Possibly move them to the functional test suite or remove if they have no value.
* [ ] Go over MigrationsInfrastructureTestBase

**Comment by ajcvickers** [comment]
See also:
* Build target model with conventions in MigrationsTestBase #20157
* Migrations: Test using snapshot #11193
